### PR TITLE
Correct setpoint target for EIRPlantLoopHeatPump::getLoadSideOutletSetPointTemp when using a dual setpoint deadband loop demand calculation scheme

### LINKED ISSUE #9739: Performance curve input for HeatPump:PlantLoop:EIR:Heating objects used with a DualSetpointDeadband plant loop demand calculation scheme is incorrect
Issue overview
--------------
When a `HeatPump:PlantLoop:EIR:Heating` object is used with a DualSetpointDeadband plant loop demand calculation scheme EnergyPlus currently uses the "high" temperature setpoint as the load side outlet setpoint. The "low" temperature setpoint should be used instead.

https://github.com/NREL/EnergyPlus/blob/4a85292a9c2c480dea9ffb11c03b8c63b16c9135/src/EnergyPlus/PlantLoopHeatPumpEIR.cc#L133-L138

### Details
Some additional details for this issue (if relevant):
 - Platform: Window
 - Version of EnergyPlus: 22.1 and 22.2

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by lymereJ**
Pull request overview
---------------------
 - Fixes #9739

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: tst/EnergyPlus/unit/PlantLoopHeatPumpEIR.unit.cc [hunk 1]
```diff
@@ -1214,6 +1214,75 @@ TEST_F(EnergyPlusFixture, CoolingOutletSetpointWorker)
...
+    // test for dual setpoint operation
+    PLHPPlantLoadSideLoop.LoopDemandCalcScheme = DataPlant::LoopDemandCalcScheme::DualSetPointDeadBand;
+    PLHPPlantLoadSideComp.CurOpSchemeType = DataPlant::OpScheme::CompSetPtBased;
+    state->dataLoopNodes->Node(thisHeatingPLHP->loadSideNodes.outlet).TempSetPointHi = 30.0;
+    state->dataLoopNodes->Node(thisHeatingPLHP->loadSideNodes.outlet).TempSetPointLo = 10.0;
+    state->dataLoopNodes->Node(5).TempSetPointHi = 30.0;
+    state->dataLoopNodes->Node(5).TempSetPointLo = 12.0;
+    EXPECT_NEAR(10.0, thisHeatingPLHP->getLoadSideOutletSetPointTemp(*state), 0.001);
+    PLHPPlantLoadSideComp.CurOpSchemeType = DataPlant::OpScheme::HeatingRB;
+    EXPECT_NEAR(12.0, thisHeatingPLHP->getLoadSideOutletSetPointTemp(*state), 0.001);
```
Comment (lymereJ) [inline tst/EnergyPlus/unit/PlantLoopHeatPumpEIR.unit.cc:1283]: New test, similar to `CoolingOutletSetpointWorker`, which checks that `HeatPump:PlantLoop:EIR:Heating` objects use the  minimum temperature setpoint.

File: src/EnergyPlus/PlantLoopHeatPumpEIR.cc [hunk 2]
```diff
@@ -132,9 +132,17 @@ Real64 EIRPlantLoopHeatPump::getLoadSideOutletSetPointTemp(EnergyPlusData &state
     } else if (thisLoadPlantLoop.LoopDemandCalcScheme == DataPlant::LoopDemandCalcScheme::DualSetPointDeadBand) {
         if (thisLoadComp.CurOpSchemeType == DataPlant::OpScheme::CompSetPtBased) {
             // there will be a valid set-point on outlet
-            return state.dataLoopNodes->Node(this->loadSideNodes.outlet).TempSetPointHi;
+            if (this->EIRHPType == DataPlant::PlantEquipmentType::HeatPumpEIRCooling) {
+                return state.dataLoopNodes->Node(this->loadSideNodes.outlet).TempSetPointHi;
+            } else {
+                return state.dataLoopNodes->Node(this->loadSideNodes.outlet).TempSetPointLo;
+            }
```
Comment (lymereJ) [inline src/EnergyPlus/PlantLoopHeatPumpEIR.cc:139]: `HeatPump:PlantLoop:EIR:Cooling` use the maximum temperature setpoint and `HeatPump:PlantLoop:EIR:Heating` use the minimum temperature setpoint.

**Comment by lymereJ** [comment]
The `DualSetpointDeadband` plant loop demand calculation scheme is to be used when the plant loop temperature is to be maintained between two temperature setpoints, a maximum and a minimum. Heating equipment should be operated to make sure that the plant loop leaving fluid temperature should not drop below the minimum temperature setpoint.

In [this](https://github.com/NREL/EnergyPlus/files/10018372/PlantLoopHeatPump_EIR_AirSource.zip) modified version of the `PlantLoopHeatPump_EIR_AirSource.idf` example file, the hot water plant loop is setup with a `DualSetpointDeadband` plant loop demand calculation scheme. In the _before_ case, the maximum temperature setpoint is used, but in the _after_ case the minimum temperature setpoint is used as expected.

image

File: src/EnergyPlus/PlantLoopHeatPumpEIR.cc (hunk 2 above)
Comment (Myoldmopar) [reply src/EnergyPlus/PlantLoopHeatPumpEIR.cc:139]: Got it, and this works for both loop and component setpoint control.  Seems like a great fix to me.

File: tst/EnergyPlus/unit/PlantLoopHeatPumpEIR.unit.cc (hunk 1 above)
Comment (Myoldmopar) [reply tst/EnergyPlus/unit/PlantLoopHeatPumpEIR.unit.cc:1283]: This test looks very reasonable.  I think you could have gotten away with not actually using any IDF syntax in this unit test, and instead just construct the heat pump directly.  Not necessary, but that's definitely a preferred direction whenever possible.

**Review by Myoldmopar — APPROVED** [review]
Code changes look good.

**Comment by Myoldmopar** [comment]
Local testing all clean, good to go.  Thanks @lymereJ

**Pull request merged**